bridge/docs/block_relay.md

```diff
@@ -0,0 +1,126 @@
+# Block Relay in Witnet
+
+The Witnet Bridge Interface (WBI) is a core component of the Witnet protocol as it enables the communication between the smart contracts in Ethereum and Witnet. In order to trustlessly verify transactions included in a Witnet block, the Ethereum smart contract needs to be able to perform verifications of the so called Proofs of Inclusion (PoI). Assuming all Witnet block headers are available to the WBI, a bridge node simply needs to provide the merkle path that includes the transaction to be verified and that reaches the root in the block header.
+
+Obviously the aforementioned mechanism works just because we assumed the block header is available in the WBI. How to make this possible with a non PoW/PoS chain is a more challenging concern though. Without loss of generality, we will assume that the WBI has access to the Witnet block headers because they are stored in another contract in Ethereum that is called the **Block Relay**. We will focus thus on how these blocks are reported to the block relay.
+
+As with the WBI, only bridge nodes have access to both Witnet and Ethereum chains. Thus, bridge nodes should be in charge of reporting valid blocks that will later be used to verify PoIs in the WBI. We observe three main challenges:
+
+- **Integrity of the Block header**: the reporting protocol needs to guarantee the integrity and validity of the block header reported.
```

Comment:
What's integrity and validity from the perspective of the relay? I'm throwing some considerations here, but I haven't arrived to any conclusion on what we should do about them:
1. Compliance of the block and transactions validation rules can't be proven directly.
2. Chain validation is possible, but AFAIK we're currently skipping such a thing (block header B that builds on block A may only get accepted if block A is known to the relay; i.e. you can not dispute the genesis block).
3. What bridge nodes can actually try to prove to the relay is broad acceptance of a particular chain of blocks.
4. However, it should not be possible for bridge nodes to fork the relay into a false chain (not considered legit by a majority of Witnet reputation power). 
5. If acceptance is proven only by majority voting (with aggregated signatures, desirably), the risk is there for an intentional or accidental breach of point 4 (relay doesn't mirror Witnet correctly) to be irreversible.
6. Reversibility of such relay failures could be achieved by means of certainty buildup over time + hard limit on reorganization depth so as to have a predictable finality with views to timely triggering contract resolution (this is indeed the typical PoW approach).

Reply:
Clearly we have two options in which the Relay can breach:

- There arent any honest bridge nodes available (assuming the majority of the ARS members are honest, one bridge node is enough to relay the correct chain)
- The attacker corrupts the majority of the ARS members.

Lets assume the latter point wont happen (a concern at Witnet protocol level, not at the bridge level). I think that, since we are making the entire set of ARS members vote, the only concern we are addressing by permiting reorganizations over a certain set of epochs would just be increasing the time-frame in which an honest bridge node is available to provide enough signatures from the ARS (specially with BLS). Which BTW is totally fine, but just to have it in mind.